terraform/modules/iam_role_policy/self_managed_mfa/policy.json

```diff
@@ -0,0 +1,49 @@
+{
+  "Version": "2012-10-17",
+  "Statement": [
+      {
+          "Sid": "AllowViewAccountInfo",
+          "Effect": "Allow",
+          "Action": "iam:ListVirtualMFADevices",
+          "Resource": "*"
+      },
+      {
+          "Sid": "AllowManageOwnVirtualMFADevice",
+          "Effect": "Allow",
+          "Action": [
+              "iam:CreateVirtualMFADevice",
+              "iam:DeleteVirtualMFADevice"
+          ],
+          "Resource": "arn:${aws_partition}:iam::*:mfa/$${aws:username}"
+      },
+      {
+          "Sid": "AllowManageOwnUserMFA",
+          "Effect": "Allow",
+          "Action": [
+              "iam:DeactivateMFADevice",
+              "iam:EnableMFADevice",
+              "iam:GetUser",
+              "iam:ListMFADevices",
+              "iam:ResyncMFADevice"
+          ],
+          "Resource": "arn:${aws_partition}:iam::*:user/$${aws:username}"
+      },
+      {
+          "Sid": "DenyAllExceptListedIfNoMFA",
```

Comment:
So this clause means people can't do _anything_ until they configure MFA.  Cool if that's what we want, but should be aware.

Reply:
This is fine by me, but good call out that we should document it.